modules/web/src/app/node-data/basic/provider/nutanix/component.ts

```diff
@@ -194,6 +284,84 @@ export class NutanixBasicNodeDataComponent extends BaseFormValidator implements
     this._cdr.detectChanges();
   }
 
+  private get _categoriesObservable(): Observable<NutanixCategory[]> {
+    return this._nodeDataService.nutanix.categories(this._clearCategory.bind(this), this._onCategoryLoading.bind(this));
+  }
+
+  private _onCategoryLoading(): void {
+    this._clearCategory();
+    this.categoryLabel = CategoryState.Loading;
+    this._cdr.detectChanges();
+  }
+
+  private _clearCategory(): void {
+    this.categories = [];
+    this._updateFilteredCategories();
+    this.categoryLabel = CategoryState.Empty;
+    this._cdr.detectChanges();
+  }
+
+  private _setCategories(categories: NutanixCategory[]): void {
+    this.categories = categories;
+    this._updateFilteredCategories();
+    this.categoryLabel = this.categories?.length ? CategoryState.Ready : CategoryState.Empty;
+    this._cdr.detectChanges();
+  }
+
+  private _updateFilteredCategories(): void {
+    const selectedCategories = this.categoriesFormArray.controls
+      .map(control => control.get(Controls.Category).value[ComboboxControls.Select])
+      .filter(Boolean);
+    selectedCategories.push('');
+    this.filteredCategories = {};
+    selectedCategories.forEach(selectedCategory => {
+      this.filteredCategories[selectedCategory] = this.categories?.filter(
+        category => category.name === selectedCategory || !selectedCategories.includes(category.name)
+      );
+    });
+  }
+
+  private _loadCategoryValues(categoryName: string): void {
+    if (categoryName && !this.categoryValuesSubscription[categoryName]) {
+      this.categoryValuesSubscription[categoryName] = this._nodeDataService.nutanix
+        .categoryValues(
+          categoryName,
+          this._clearCategoryValue.bind(this, categoryName),
+          this._onCategoryValueLoading.bind(this, categoryName)
+        )
+        .pipe(takeUntil(this._unsubscribe))
+        .subscribe(this._setCategoryValues.bind(this, categoryName));
+    }
+  }
+
+  private _onCategoryValueLoading(categoryName: string): void {
+    this._clearCategoryValue(categoryName);
+    this.categoryValuesLabel[categoryName] = CategoryValueState.Loading;
+    this._cdr.detectChanges();
+  }
+
+  private _clearCategoryValue(categoryName: string): void {
+    this.categoryValues[categoryName] = [];
+    this.categoryValuesLabel[categoryName] = CategoryValueState.Empty;
+    this._cdr.detectChanges();
+  }
+
+  private _setCategoryValues(categoryName: string, values: NutanixCategoryValue[]): void {
+    this.categoryValues[categoryName] = values;
```

Comment:
I think this is more sort of BE thing but if backend is returning and following labels needs to be filtering in case of Edit MD dialog since shared component is used in
- ClusterList Page -> edit MD from list 
OR
- MD details Page -> edit MD 

I created Cluster with following Categories (2 categories)
On Cluster and Nodes creation if you try to edit MD w.r.t this image 4 calls are sent for each category since 2 extra categories are received in cloud spec.


**Categories added while creating**

![1](https://user-images.githubusercontent.com/17727069/202122635-9db529ac-7ffd-44d9-80f3-415f1a66d32e.png)


**In case edit MD dialog**
![2](https://user-images.githubusercontent.com/17727069/202123841-ce7cd35b-8c50-4b82-b49d-31c2a3157842.png)

**MD details page**
![2extra](https://user-images.githubusercontent.com/17727069/202124410-139dc24b-fae2-4f3c-975f-fbb970088b62.png)

Reply:
Those `KKP` categories will be hidden from API response as per my discussion with @embik , so no change is required on frontend.